src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -47,59 +75,152 @@ private CoseHeaderMap(bool isReadOnly)
         public void Add(CoseHeaderLabel key, CoseHeaderValue value)
         {
             ValidateIsReadOnly();
             ValidateInsertion(key, value);
             _headerParameters.Add(key, value);
         }
 
+        /// <summary>
+        /// Adds the specified value to the <see cref="ICollection{T}"/> with the specified key.
+        /// </summary>
+        /// <param name="item">The <see cref="KeyValuePair{CoseHeaderLabel, CoseHeaderValue}"/> structure representing the key and value to add to the <see cref="CoseHeaderMap"/>.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
+        /// <exception cref="ArgumentException">The property is set and the encoded bytes in the specified <see cref="CoseHeaderValue"/> contain trailing data or more than one CBOR value.</exception>
         public void Add(KeyValuePair<CoseHeaderLabel, CoseHeaderValue> item) => Add(item.Key, item.Value);
 
+        /// <summary>
+        /// Adds the specified key and value to the <see cref="CoseHeaderMap"/>. The specified value will be encoded as CBOR and stored as a <see cref="CoseHeaderValue"/>.
+        /// </summary>
+        /// <param name="label">The key of the element to add.</param>
+        /// <param name="value">The value of the element to encode and then add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
         public void Add(CoseHeaderLabel label, int value) => Add(label, CoseHeaderValue.FromInt32(value));
 
+        /// <summary>
+        /// Adds the specified key and value to the <see cref="CoseHeaderMap"/>. The specified value will be encoded as CBOR and stored as a <see cref="CoseHeaderValue"/>.
+        /// </summary>
+        /// <param name="label">The key of the element to add.</param>
+        /// <param name="value">The value of the element to encode and then add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
         public void Add(CoseHeaderLabel label, string value) => Add(label, CoseHeaderValue.FromString(value));
 
+        /// <summary>
+        /// Adds the specified key and value to the <see cref="CoseHeaderMap"/>. The specified value will be encoded as CBOR and stored as a <see cref="CoseHeaderValue"/>.
+        /// </summary>
+        /// <param name="label">The key of the element to add.</param>
+        /// <param name="value">The value of the element to encode and then add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
         public void Add(CoseHeaderLabel label, byte[] value) => Add(label, CoseHeaderValue.FromBytes(value));
```

Comment:
Because bytes are ambiguous, this member (and the ROSpan one) should mention encoding.  But in a remarks section, not the summary.

```suggestion
        /// <remarks>
        /// <paramref name="value"/> does not need to contain a valid CBOR-encoded value, as it will be wrapped in a CBOR <c>bstr</c> encoding.
        /// To specify a CBOR-encoded value directly, see <see cref="CoseHeaderValue.FromEncodedValue" /> and <see cref="Add(CoseHeaderLabel, CoseHeaderValue)" />.
        /// </remarks>
        public void Add(CoseHeaderLabel label, byte[] value) => Add(label, CoseHeaderValue.FromBytes(value));
```

Reply:
How about "... as it will be encoded as a CBOR byte string."?

Reply:
Instead of "as it will be wrapped..."?  Sure, that works.